cli/Sources/TuistGenerator/Extensions/TargetDependency+Extras.swift

```diff
@@ -0,0 +1,12 @@
+import XcodeGraph
+
+extension TargetDependency {
+    var hasSignature: Bool {
+        switch self {
+        case let .xcframework(_, expectedSignature, _, _):
+            expectedSignature != nil
+        default:
+            false
+        }
+    }
+}
```

Comment:
I'd recommend not using `Target.dependencies`. Those values are populated when the manifests are loaded, and then mapped to `Graph.dependencies`, which can then be mutated. Since the `generate` function takes a `GraphTraverser`, you can expose an interface there along the lines of:

```swift
graphTraverser.haveProjectTargetsDependenciesWithSignatures(projectPath)
```

And then the implementation of the function can use `graph.dependencies` as the source of truth.